src/middlewares/json-api-vercel.ts

```diff
@@ -0,0 +1,45 @@
+import { Operation, VercelRequest, VercelResponse } from "../types";
+
+import Application from "../application";
+import ApplicationInstance from "../application-instance";
+
+import {
+  authenticate,
+  urlData,
+  handleBulkEndpoint,
+  handleJsonApiEndpoint,
+  convertErrorToHttpResponse
+} from "../utils/http-utils";
+
+export default function jsonApiVercel(app: Application) {
+  return async (req: VercelRequest, res: VercelResponse) => {
+    const appInstance = new ApplicationInstance(app);
+
+    try {
+      await authenticate(appInstance, req);
+    } catch (error) {
+      res.status(error.status);
+      res.json(convertErrorToHttpResponse(error));
+    }
+
+    if (process.env.AWS_REGION) {
+      // This means we're not in localhost.
+      req["href"] = `https://${req.headers.host}${req.url!}`;
+    } else {
+      // This means we're in development. We won't use HTTPS here.
+      req["href"] = `http://localhost${req.url!}`;
+    }
+
+    req["urlData"] = urlData(appInstance, req.url!);
```

Comment:
It seems like `req.body` is a `string` on POST requests. I think it's because of nonstandard `
content-type:'application/vnd.api+json'` and Next.js don't know how to parse it.

Reply:
I had to do this it make it work
```
export default (req: NextApiRequest, res: NextApiResponse) => {
	if (req.body) {
		req.body = JSON.parse(req.body);
	}

	jsonApiVercel(app)(req, res);
};
```

Reply:
Good catch! I'll add a conversion for it on the `jsonApiNext` middleware.

Reply:
I just realized that this is because my client lib sends jsonapi Content-Type on every request. 
I have to check what happens if I don’t send Content-Type.

Reply:
If I don't send a content-type Next detect body as `content-type:'text/plain;charset=UTF-8'` and body is a string.
If I set `'content-type': 'application/json'`  then `body` is automatically parsed to object by Next.js.
Finally, if I set `'content-type': 'application/vnd.api+json'` body will be string again.

Should Kurier only allow `'content-type': 'application/vnd.api+json'`?
https://jsonapi.org/format/1.1/#introduction
> JSON:API requires use of the JSON:API media type (application/vnd.api+json) for exchanging data.

Reply:
While I agree that the required type is `vnd.api+json`, we wanted to make adoption easier and not change a thing in the frontend side. That is -- if you're already using `application/json`, we should be able to work with that.

Reply:
Maybe we could add `strict` mode and throw an error if `content-type` isn't correct...

Reply:
I like it. Such setting would live in the `Application` options, I imagine.